mixnet/client/src/lib.rs

```diff
@@ -24,8 +24,17 @@ impl<R: Rng> MixnetClient<R> {
         }
     }
 
-    pub async fn run(&self, message_tx: impl Sink<Vec<u8>> + Clone + Unpin + Send + 'static) {
-        self.mode.run(message_tx).await
+    // Instead of providing this sync function that spawns a task internally,
+    // it would be better to provide the `Stream` trait implementation
+    // for better composability from the user's point of view.
+    // https://github.com/logos-co/nomos-node/issues/314
+    pub fn run(&self, message_tx: impl Sink<Vec<u8>> + Clone + Unpin + Send + Sync + 'static) {
+        let mode = self.mode.clone();
+        tokio::spawn(async move {
+            if let Err(e) = mode.run(message_tx).await {
+                tracing::error!("error from mixnet client receiver: {e}");
+            }
+        });
```

Comment:
This is what I don't like. As written in the comment, we may need to refactor this.

I made this change to avoid the borrow error when integrating the previous code with the libp2p backend:
```rust
error[E0382]: use of moved value: `mixnet_client`
   --> nomos-services/network/src/backends/libp2p.rs:101:42
    |
88  |           let mut mixnet_client = MixnetClient::new(config.mixnet_client, OsRng);
    |               ----------------- move occurs because `mixnet_client` has type `MixnetClient<OsRng>`, which does not implement the `Copy` trait
...
91  |           overwatch_handle.runtime().spawn(async move {
    |  __________________________________________-
92  | |             mixnet_client.run(PollSender::new(mixnet_inbound_tx)).await;
    | |             ------------- variable moved due to use in generator
93  | |         });
    | |_________- value moved here
...
101 |           overwatch_handle.runtime().spawn(async move {
    |  __________________________________________^
102 | |             use tokio_stream::StreamExt;
103 | |             let mut swarm = Swarm::build(&config.inner).unwrap();
104 | |             loop {
...   |
153 | |                                 log_error!(mixnet_client.send(msg));
    | |                                            ------------- use occurs due to use in generator
...   |
204 | |             }
205 | |         });
    | |_________^ value used here after move
```